### PR TITLE
Improved GET /events/find call with Complex Logic

### PR DESCRIPTION
**Pull request by israeltq**
Improved our findEvents handler to return most relevant events for a particular user in addition to the events that the user is part of already (i.e. he created, was invited to, is going to)

### CODE REVIEW

File: api/handlers/event.js [hunk 1]
```diff
@@ -254,13 +257,36 @@ let findEvents = async function(req, resp) {
         public: true
     };
 
-    Event.find(query, function(err, events) {
-        if (err) {
-            console.log(`[error] ${err}`);
-            return resp.status(500).send("Could not retrieve events");
+    if (categories && categories.length != 0) {
+        nearEventsQuery.categories = { $all: categories };
```
Comment (adrianosela) [inline api/handlers/event.js]: nice

File: api/handlers/event.js [hunk 2]
```diff
@@ -299,6 +325,55 @@ let searchEvents = async function(req, res) {
...
+        }).indexOf(bestEvent._id);
+        events.splice(index, index + 1);
+
+        relevant.push(bestEvent);
+    }
+
+    return relevant;
+};
+
+let friendsGoingToEvent = function(user, event) {
```
Comment (adrianosela) [inline api/handlers/event.js]: ideally we'd give the db query a set of users and perhaps theres an $any keyword or smth

File: api/handlers/event.js [hunk 3]
```diff
@@ -299,6 +325,55 @@ let searchEvents = async function(req, res) {
     }
 };
 
+// TODO: Define a default limit of returned events
+let getRelevantEventsForUser = function(events, user, limit = 10) {
```
Comment (adrianosela) [inline api/handlers/event.js]: this could all be done with the db query

**Comment by adrianosela** [comment]
We can do the search programmatically for now. this is good. But ideally we'd be doing it just with the db query.

File: api/handlers/event.js (hunk 3 above)
Comment (israeltq) [reply api/handlers/event.js]: In theory yes, but the point of the complex logic is that we do it ourselves

**Pull request merged**